worlds/oot_soh/location_access/dungeons/forest_temple.py

```diff
@@ -2,8 +2,465 @@
+    add_locations(Regions.FOREST_TEMPLE_LOBBY, world, [
+        (Locations.FOREST_TEMPLE_GS_LOBBY, lambda bundle: hookshot_or_boomerang(bundle)),
+        (Locations.FOREST_TEMPLE_LOBBY_POT1, lambda bundle: can_break_pots(bundle)),
+        (Locations.FOREST_TEMPLE_LOBBY_POT2, lambda bundle: can_break_pots(bundle)),
+        (Locations.FOREST_TEMPLE_LOBBY_POT3, lambda bundle: can_break_pots(bundle)),
+        (Locations.FOREST_TEMPLE_LOBBY_POT4, lambda bundle: can_break_pots(bundle)),
+        (Locations.FOREST_TEMPLE_LOBBY_POT5, lambda bundle: can_break_pots(bundle)),
+        (Locations.FOREST_TEMPLE_LOBBY_POT6, lambda bundle: can_break_pots(bundle)),
+    ])
+    # Connections
+    connect_regions(Regions.FOREST_TEMPLE_LOBBY, world, [
+        (Regions.FOREST_TEMPLE_SOUTH_CORRIDOR, lambda bundle: True),
+        (Regions.FOREST_TEMPLE_NORTH_CORRIDOR, lambda bundle: True),
+        (Regions.FOREST_TEMPLE_NW_OUTDOORS_LOWER, lambda bundle: (can_use(Items.SONG_OF_TIME, bundle) or is_child(bundle))),
+        (Regions.FOREST_TEMPLE_NE_OUTDOORS_LOWER, lambda bundle: (can_use(Items.FAIRY_BOW, bundle) or can_use(Items.FAIRY_SLINGSHOT, bundle))),
+        (Regions.FOREST_TEMPLE_WEST_CORRIDOR, lambda bundle: (small_keys(Items.FOREST_TEMPLE_SMALL_KEY, 1, bundle))),
+        (Regions.FOREST_TEMPLE_EAST_CORRIDOR, lambda bundle: False),
+        (Regions.FOREST_TEMPLE_BOSS_REGION, lambda bundle: has_item(LocalEvents.DEFEATED_MEG, bundle)),
+        (Regions.FOREST_TEMPLE_BOSS_ENTRYWAY, lambda bundle: False)
+    ])
+    
+    ## Forest Temple North Corridor
+    # Connections
+    connect_regions(Regions.FOREST_TEMPLE_NORTH_CORRIDOR, world, [
+        (Regions.FOREST_TEMPLE_LOBBY, lambda bundle: True),
+        (Regions.FOREST_TEMPLE_LOWER_STALFOS, lambda bundle: True)
+    ])
+    
+    ## Forest Temple Lower Stalfos
+    # Events
+    add_events(Regions.FOREST_TEMPLE_LOWER_STALFOS, world, [
+        (EventLocations.FOREST_TEMPLE_LOWER_STALFOS_FAIRY_POT, LocalEvents.LOWER_STALFOS_FAIRY, lambda bundle: True),
+    ])
+    # Locations
+    add_locations(Regions.FOREST_TEMPLE_LOWER_STALFOS, world, [
+        (Locations.FOREST_TEMPLE_FIRST_STALFOS_CHEST, lambda bundle: (can_kill_enemy(bundle, Enemies.STALFOS, EnemyDistance.CLOSE, True, 2))),
+        (Locations.FOREST_TEMPLE_LOWER_STALFOS_POT1, lambda bundle: can_break_pots(bundle)),
+        (Locations.FOREST_TEMPLE_LOWER_STALFOS_POT2, lambda bundle: can_break_pots(bundle)),
+    ])
+    # Connections
+    connect_regions(Regions.FOREST_TEMPLE_LOWER_STALFOS, world, [
+        (Regions.FOREST_TEMPLE_NORTH_CORRIDOR, lambda bundle: True)
+    ])
+    
+    ## Forest Temple NW Outdoors Lower
+    # Events
+    add_events(Regions.FOREST_TEMPLE_NW_OUTDOORS_LOWER, world, [
+        (EventLocations.FOREST_TEMPLE_STICKS, Events.CAN_FARM_STICKS, lambda bundle: can_get_deku_baba_sticks(bundle)),
+        (EventLocations.FOREST_TEMPLE_NUTS, Events.CAN_FARM_NUTS, lambda bundle: can_get_deku_baba_nuts(bundle))
+    ])
```

Comment:
These EventLocations also need to be renamed to reflect the region they're in. `FOREST_TEMPLE_NW_OUTDOORS_LOWER_DEKU_BABA_STICKS` etc. I'm also not seeing these events in the EventLocations enum to begin with.